extinguish.lua

```diff
@@ -0,0 +1,187 @@
+-- Puts out fires.
+-- author: Atomic Chicken
+
+local utils = require 'utils'
+local validArgs = utils.invert({
+'all',
+'location',
+'help'
+})
+local args = utils.processArgs({...}, validArgs)
+
+--@ module = true
```

Comment:
Personally, I'd move this after the comments at the top of the file, but this is close enough.